An example OSGi bundle

OndroMih opened the pull request.

mulderbaba commented:
I agree on @smillidge but I'm totally against the reverse, having modules know their parents.

pdudits commented on OSGi/bundle-with-package-dependency/pom.xml:
```diff
@@ -0,0 +1,50 @@
+                        <Bundle-Activator>fish.payara.bundle.Activator</Bundle-Activator>
+                        <Import-Package>org.eclipse.persistence.expressions,
+                            org.osgi.framework;version="[1.5,2)"</Import-Package>
+                        <Bundle-SymbolicName>Payara-Example</Bundle-SymbolicName>
```
It is recommended to treat symbolic name like a package (or automatic module name in recent days), therefore `fish.payara.bundle` is more fitting name

OndroMih replied:
I changed both package name and symbolic name to `fish.payara.examplebundle`

OndroMih commented:
> Should this be linked into the parent pom as a module?
I added the osgi module to parent pom. The osgi module doesn't refer to the parent pom and can still be built separately.

I also renamed the OSGi folder to osgi and added a parent osgi module.
